applications/convection_diffusion_application/custom_elements/laplacian_element.cpp

```diff
@@ -73,51 +75,54 @@ void LaplacianElement::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix, Vec
     noalias(rRightHandSideVector) = ZeroVector(number_of_points); //resetting RHS
 
     //reading integration points and local gradients
-    const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints();
-    const GeometryType::ShapeFunctionsGradientsType& DN_De = GetGeometry().ShapeFunctionsLocalGradients();
-    const Matrix& N_gausspoint = GetGeometry().ShapeFunctionsValues();
+    const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints();
+    const GeometryType::ShapeFunctionsGradientsType& DN_De = r_geometry.ShapeFunctionsLocalGradients();
+    const Matrix& N_gausspoint = r_geometry.ShapeFunctionsValues();
 
     Element::GeometryType::JacobiansType J0;
     Matrix DN_DX(number_of_points,dim);
     Matrix InvJ0(dim,dim);
     Vector temp(number_of_points);
 
     Vector heat_flux_local(number_of_points);
+    Vector nodal_conductivity(number_of_points);
     for(unsigned int node_element = 0; node_element<number_of_points; node_element++)
     {
-        heat_flux_local[node_element] = GetGeometry()[node_element].FastGetSolutionStepValue(HEAT_FLUX);
+        heat_flux_local[node_element] = r_geometry[node_element].FastGetSolutionStepValue(HEAT_FLUX);
+        nodal_conductivity[node_element] = r_geometry[node_element].FastGetSolutionStepValue(CONDUCTIVITY);
```

Comment:
Maybe if we do it as in the `eulerian_conv_diff` element we can reuse this same element for other "Laplacian" purposes. I mean to retrieve the conductivity variable from the convection diffusion settings.

What do you think?

Reply:
I thought about that, but the convection-diffusion settings have a lot of variables that would never be used here... We should consider using it, but maybe throwing errors if the unsupported variables (convection velocity, specific heat...) are defined.

Reply:
OK then. Let's leave it for another PR.